addons/mail/static/src/models/attachment_link_preview_view.js

```diff
@@ -0,0 +1,108 @@
+/** @odoo-module **/
+
+import { registerModel } from '@mail/model/model_core';
+import { attr, one } from '@mail/model/model_field';
+import { clear, insertAndReplace } from '@mail/model/model_field_command';
+
+registerModel({
+    name: 'AttachmentLinkPreviewView',
+    identifyingFields: ['attachmentList', 'attachment'],
+    recordMethods: {
+        /**
+         * Handles the click on delete attachment and open the confirm dialog.
+         *
+         * @param {MouseEvent} ev
+         */
+        onClickUnlink(ev) {
+            if (!this.attachment) {
+                return;
+            }
+            if (this.attachmentList.composerViewOwner) {
+                this.attachment.remove();
+            } else {
+                this.update({ attachmentDeleteConfirmDialog: insertAndReplace() });
+            }
+        },
+        /**
+         * @private
+         * @returns {number|FieldCommand}
+         */
+        _computeHeight() {
+            if (!this.attachment) {
+                return clear();
+            }
+            if (this.attachment.composer) {
+                return 50;
+            }
+            return 160;
+        },
+        /**
+         * @private
+         * @returns {string|FieldCommand}
+         */
+        _computeImageUrl() {
+            if (!this.attachment) {
+                return clear();
+            }
+            if (!this.attachment.accessToken && this.attachment.originThread && this.attachment.originThread.model === 'mail.channel') {
+                return `/mail/channel/${this.attachment.originThread.id}/image/${this.attachment.id}/${this.width}x${this.height}?checksum=${this.attachment.checksum}`;
+            }
+            const accessToken = this.attachment.accessToken ? `?access_token=${this.attachment.accessToken}&checksum=${this.attachment.checksum}` : `?checksum=${this.attachment.checksum}`;
+            return `/web/image/${this.attachment.id}/${this.width}x${this.height}${accessToken}`;
+        },
+        /**
+         * @private
+         * @returns {number|FieldCommand}
+         */
+        _computeWidth() {
+            if (!this.attachment) {
+                return clear();
+            }
+            return 160;
+        },
```

Comment:
All methods should guard `!this.exists()` instead of `!this.attachment`.
When the record exists, `this.attachment` always exists because it's required identifying field.